### DISCUSSION TITLE
consensus/Multi-Paxos

### DISCUSSION THREAD

**Discussion by giscus** [isno/theByteBook#178] [General]
# consensus/Multi-Paxos

构建大规模高可用的分布式系统

https://www.thebyte.com.cn/consensus/Multi-Paxos.html

**Comment by littlebear-xbz** [comment]
第一段这里Multi 是否指的是 Basic

lamport 提到的 Multi Paxos 是一种思想，所以 Multi Paxos 算法实际上是个统称，Multi Paxos 算法是指基于 **Multi** Paxos 思想，通过多个 Basic Paxos 实例实现的一系列值的共识算法。

**Reply by isno** [reply]
是的，这一段写点有点绕了。我回来改改

**Comment by madeinchinalmc** [comment]
下面我们举个例子说明，当 S1 收到客户端的请求命令 jmp（提案）时（假设此时是 S3 宕机）： ... S1 继续尝试下一个日志项，直到给 jmp 找到一个可以达成共识的位置（日志项 4）。 这里的应该是“直到给 jmp 找到一个可以达成共识的位置（日志项 5）”

**Comment by madeinchinalmc** [comment]
另外一个思路就是优化二阶段提交的次数，具体的方式是减小 Prepare 请求之前 这句话中“减小 Prepare 请求之前” 感觉这句话不太完整，减少请求之前的时间？

**Reply by isno** [reply]
共识的内容，我在重新编写。请过一段时间再看。或者等到纸质书籍出版再看。
